pkg/api/cached.go

```diff
@@ -167,11 +174,74 @@ func (c *Cached) NodePublishVolume(ctx context.Context, req *csi.NodePublishVolu
 		return nil, fmt.Errorf("failed to change ownership of target directory %s: %s", targetPath, err)
 	}
 
-	version, err := c.writeCache(cachePath)
-	if err != nil {
-		return nil, err
-	}
+	// Perform an overlay mount if the mountCache attribute is true
+	// Mount the root `StagingPath` at `targetPath/gadget` making `targetPath/gadget` `0777`
+	if mountCache, exists := volumeAttributes["mountCache"]; exists && mountCache == "true" {
+		upperdir := path.Join(targetPath, "gadget_writeable")
+		err := os.MkdirAll(upperdir, 0777)
+		if err != nil {
+			return nil, fmt.Errorf("failed to create gadget_writeable directory %s: %v", upperdir, err)
+		}
+
+		workdir := path.Join(targetPath, "work")
+		err = os.MkdirAll(workdir, 0777)
+		if err != nil {
+			return nil, fmt.Errorf("failed to create overlay work directory %s: %v", workdir, err)
+		}
+
+		// Create the cache directory, we can make it writable by the pod
+		gadgetDir := path.Join(targetPath, "gadget")
+		err = os.MkdirAll(gadgetDir, 0777)
+		if err != nil {
+			return nil, fmt.Errorf("failed to create gadget directory %s: %v", gadgetDir, err)
+		}
 
+		mountArgs := []string{
+			"-t",
+			"overlay",
+			"overlay",
+			gadgetDir,
+			"-n",
+			"-o",
+			fmt.Sprintf("lowerdir=%s,upperdir=%s,workdir=%s", c.StagingPath, upperdir, workdir),
```

Comment:
We should consider what value we want for the `redirect_dir` option. When I was working with overlayfs for making the gadget CI bits that reset the filesystem faster, I had to set it to `on`, but that's not necessarily true here. https://docs.kernel.org/filesystems/overlayfs.html#redirect-dir

See https://github.com/gadget-inc/gadget/pull/11070/files#diff-37490d715785dc29d4323daef03e84518b414f5fdca4d258c9f415c936080d4f if you are curious

Reply:
TIL. 
We can set it, I'm not sure how often we'll change the cache but it'll be nice to have the option.

What do you mean by 
> CI bits that reset the filesystem faster

As in, `overlay` was faster or something else?